src/poetry/console/commands/init.py

```diff
@@ -270,10 +271,14 @@ def _determine_requirements(
                 else:
                     choices = []
                     matches_names = [p.name for p in matches]
-                    exact_match = constraint["name"] in matches_names
+                    exact_match = canonicalize_name(constraint["name"]) in matches_names
```

Comment:
It would be good to use a local variable such as `canonicalized_name` instead of calling the helper twice.